design/ErrorHandling.md

```diff
@@ -184,3 +163,21 @@ It is only necessary if an application wishes to recover from recoverable errors
  - Should an object creation error immediately log an error to the error log?
    Or should it only log if the error propagates to a device-level operation?
     - Tentatively resolved: errors should be logged immediately.
+
+ - Should applications be able to intentionally create graphs of potentially-invalid objects, and recover from this late?
+   E.g. create a large buffer, create a bind group from that, create a command buffer from that, then choose whether to submit based on whether the buffer was successfully allocated.
+    - If yes, `tryCreateBuffer` must return `GPUBuffer` and error log entries must not be generated when creating objects from invalid objects.
+      (Only log errors on queue.submit and other device/queue level operations.)
+    - If no, `tryCreateBuffer` should return `Promise<GPUBuffer?>` and error log entries should be generated when creating objects from invalid objects.
+    - Tentatively resolved: no.
+
+ - Should there be an API to query object status?
+    - Tentatively resolved: no. (API was removed in #197.)
+       - This query was only useful for detecting createPipeline completion and createBuffer/Texture OOM (now both done via Promise).
+       - Validation errors should not be detected, because they indicate programming errors (hopefully), so they are surfaced through the error log instead.
+       - Device loss applies to all objects on a device, and is handled through a separate mechanism, so it's not useful to separately know that an individual object on the device is invalid.
+
+ - Should `tryCreate*` (and `createReady*Pipeline`) resolve to invalid objects on validation failure and device loss, instead of rejecting?
+   This simplifies things slightly by avoiding giving useless extra info to the app via "reject" (which they would ignore/noop anyway).
+   Instead these Promises would never reject.
+    - Tentatively resolved: yes.
```

Comment:
If there are any further comments on my proposed resolve/reject semantics of tryCreate/createReady, this thread would be a good place.